sstables/sstable_directory.cc

```diff
@@ -248,12 +248,13 @@ future<> sstable_directory::components_lister::process(process_flags flags, fs::
     }
 }
 
-future<>
-sstable_directory::commit_directory_changes() {
-    auto files_for_removal = std::move(_lister->_state->files_for_removal);
-    _lister.reset();
+future<> sstable_directory::commit_directory_changes() {
+    return _lister->commit().finally([x = std::move(_lister)] {});
```

Comment:
````
    auto lister = std::move(_lister);
    co_await lister->commit();
````

Reply:
Yup, looks much nicer indeed. I'll prepare the coroutinization PR on sstable directory as a followup then